### ISSUE TITLE
Cannot use `useI18n` composable inside Nuxt 3 plugin

### ISSUE THREAD

**Issue by F0rsaken** [nuxt-modules/i18n#2484]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.16.0`
- Nuxt Version:     `3.7.4`
- CLI Version:      `3.9.0`
- Nitro Version:    `2.6.3`
- Package Manager:  `yarn@1.22.19`
- Builder:          `-`
- User Config:      `runtimeConfig`, `ssr`, `devServer`, `nitro`, `vite`, `app`, `css`, `modules`, `cleanupAliases`, `i18n`, `dayjs`, `build`, `hooks`
- Runtime Modules:  `normalizedModule()`, `@nuxtjs/i18n@8.0.0-rc.5`, `dayjs-nuxt@2.0.2`, `@vueuse/nuxt@10.4.1`, `@pinia/nuxt@0.4.11`
- Build Modules:    `-`

### Reproduction

To reproduce this issue, just install `@nuxt/i18n` package in `v8` version, create a plugin and use `useI18n` composable.

Reproduction sandbox: https://codesandbox.io/p/sandbox/winter-architecture-vxk7f3

### Describe the bug

Composable `useI18n` throws `Must be called at the top of a 'setup' function` when used in context when composables are usable.

You can work around it with `useNuxtApp().$i18n`, but it would be nice if this composable could be used from the start with package. I could make a PR with such composable, if I'm not the only one, who wants something like that.

**Comment by kazupon** [comment]
Thank you for your feedback.

`useI18n` is Composition API of vue-i18n.
The Composition API has rules for use in setup (or <script setup>`).
The same is true for Nuxt.
https://vuejs.org/api/composition-api-setup.html

I don't know your use case, but if you want to use it in nuxt pluign, you can use `useNuxtApp(). $i18n`.

Thanks!

**Comment by lukaszflorczak** [comment]
@kazupon I have doubts about whether relying solely on the Vue documentation is justified. Composables can be used in Nuxt plugins or even in server routes. Vue doesn't have such elements, so it naturally doesn't mention them, focusing mainly on components. 

Regarding plugins in Nuxt, there are, of course, certain limitations mentioned in the documentation: https://nuxt.com/docs/guide/directory-structure/plugins#using-composables

The question is whether such limitations actually exist in the case of i18n, if we can easily use `useNuxtApp().$i18n`. In other modules (e.g., Pinia), `usePinia()` is equivalent to `$pinia`, so it would be nice if this could be standardized as much as possible.

Now it seems like using the composable outside of the `setup` is artificially blocked.

**Comment by kazupon** [comment]
@lukaszflorczak 
Thank you for your feedback! ❤️ 

I think it would be great if we could use `useI18n` with the nuxt plugin too.
However, `useI18n` is already designed to work with Vue lifetime and is provided as composable.

There are two types of `useI18n` for use within Vue components: global scope, which allows application-wide use of i18n functionality, and local scope, which allows resource and locale management on a per Vue component.

It's possible to extend (wrap) `useI18n` to export as a composable that runs outside of a setup like the nuxt plugin (pinia). In that case, it will only work with global scope.
As drawbacks where `useI18n` can be used anywhere, we need to be aware of the context in which the composable is called.

I am concerned about the cognitive load I would place on users who are already using `useI18n`.

As far as I found out, `usePinia` is a composable exported by the `@pinia/nuxt` module.

That composable was added when pinia released nuxt module for nuxt.
I think that pinia did not need to consider existing API when `@pinia/nuxt` has shiped.
